mathewDavidov/assignments/slidingwindow/lc209/lc209.java

```diff
@@ -0,0 +1,52 @@
+/*
+Given an array of positive integers nums and a positive integer target, 
+return the minimal length of a contiguous subarray [numsl, numsl+1, ..., numsr-1, numsr] 
+of which the sum is greater than or equal to target. If there is no such subarray, return 0 instead.
+
+Input: target = 7, nums = [2,3,1,2,4,3]
+Output: 2
+Explanation: The subarray [4,3] has the minimal length under the problem constraint.
+
+edge:
+arr.length > 0 || null
+target > 0
+
+currentSum, windowStart, minLength
+for 0 - end
+    currentSum += arr[windowEnd]
+
+    while currentSum >= target
+        minLength = min(minLength, windowEnd-windowStart+1)
+        currentSum -= arr[windowStart]
+        windowStart++;
+*/
```

Comment:
It is great you wrote some psuedo-code out. To make this even better, you could write it as sentences in the event that coming up with a coded solution becomes difficult. That way the interviewer will understand your approach without having to translate the code written out.